portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -22,44 +33,65 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that returns some placeholder comment saying 'Hello, Omar!'. TODO(oumontiel): modify this file to handle comments data */
+/** Servlet that posts and retrieves comments from Datastore. */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
   /**
-   * List of messages to be fetched from this Data Servlet
-   * @private {ArrayList<String>}
+   * Gets the comments, in JSON format, taken from user input in the form from the HTML,
+   * which are stored in the messages variable.
    */
-  private ArrayList<String> messages;
-
   @Override
-  public void init() {
-    messages = new ArrayList<>();
-    messages.add("Once upon a time...");
-    messages.add("Twice upon two times...");
-    messages.add("Thrice upon three times...");
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    /** Create a query and prepare it with the data stored in Datastore. */
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery preparedQuery = datastore.prepare(query);
+
+    /** Set maximum number of comments to be included in the response. */
+    int numOfComments = Integer.parseInt(request.getParameter("limit"));
+    List<Entity> results = preparedQuery.asList(FetchOptions.Builder.withLimit(numOfComments));
+
```

Comment:
nit: two blank lines are permitted but not encouraged - https://google.github.io/styleguide/javaguide.html#s4.6.1-vertical-whitespace

Reply:
Done!